gobblin-runtime/src/main/java/org/apache/gobblin/runtime/GobblinMultiTaskAttempt.java

```diff
@@ -343,8 +354,8 @@ private boolean taskSuccessfulInPriorAttempt(String taskId) {
       // Create a new task from the work unit and submit the task to run
       Task task = createTaskRunnable(workUnitState, countDownLatch);
       this.taskStateTracker.registerNewTask(task);
+      task.setTaskFuture(this.taskExecutor.submit(task));
```

Comment:
Also, if we are going to "require" a future, we should probably remove the `TaskExecutor#execute()` methods and figure out how to ensure that a submitted task always has the future set on it without requiring the caller to do it explicitly.

Reply:
Why do we not want TaskExecutor#execute() along submit() ?
Both methods were present since the beginning.
taskExecutor#execute() is not being used in gobblin anywhere, and can be removed safely. But there is a possibility some forked branch is using it.

Reply:
Right, but seems like that method is not work correctly going forward.  Unless there is a compelling reason to keep it, it seems like we shouldn't have a method there that "might" be used.  Then we have to test it and maintain it.  That is one of the main challenges with Gobblin at the moment.

Reply:
No no, execute() does work correctly, but execute() does not return a Future, and we need a handle to the Future in this case. In one of the Gobblin based internal project of LinkedIn, we used execute()

Reply:
It only works correctly if not used in the scope of Helix.  Either way, but we need to work to a very clear interface that both internal LinkedIn projects and external projects use.  This will ensure clarity and reduce the test surface.